File: prototype/cpp/shape.cpp
#ifndef shape_impl
#define shape_impl

#include "shape.hpp"
#include <string>

void Shape::setColor(string new_color)
{
  color = new_color;
};

void Shape::setX(int new_x)
{
  x = new_x;
};

void Shape::setY(int new_y)
{
  y = new_y;
};

int Shape::getX()
{
  return x;
};

int Shape::getY()
{
  return y;
};

string Shape::getColor()
{
  return color;
}

#endif

File: Generating/prototype/rust/shape.rs
#[derive(Debug, Clone)]
pub struct Shape {
  pub x: i32,
  pub y: i32,
  pub color: String
}

pub trait ShapeTrait {
  fn clone_instance(&self) -> Self;
  fn set_x(&mut self, x: i32);
  fn set_y(&mut self, x: i32);
  fn set_color(&mut self, color: &str);
}